include/seqan3/alphabet/cigar/cigar.hpp

```diff
@@ -0,0 +1,326 @@
+
+    using base_type::operator=; // Inherit non-default assignment operators
+
+    //!\copydoc cartesian_composition::cartesian_composition(component_type const alph)
+    SEQAN3_DOXYGEN_ONLY(( constexpr cigar(component_type const alph) {} ))
+    //!\copydoc cartesian_composition::cartesian_composition(indirect_component_type const alph)
+    SEQAN3_DOXYGEN_ONLY(( constexpr cigar(indirect_component_type const alph) {} ))
+    //!\copydoc cartesian_composition::operator=(component_type const alph)
+    SEQAN3_DOXYGEN_ONLY(( constexpr cigar & operator=(component_type const alph) {} ))
+    //!\copydoc cartesian_composition::operator=(indirect_component_type const alph)
+    SEQAN3_DOXYGEN_ONLY(( constexpr cigar & operator=(indirect_component_type const alph) {} ))
+    //!\}
+
+    /*!\name Write functions
+     * \{
+     */
+    //!\brief Assign from a character. This modifies the internal sequence letter.
+    constexpr cigar & assign_char(char_type const c)
+    {
+        seqan3::assign_char(get<0>(*this), c);
+        return *this;
+    }
+
+    //!\brief Assign from a count value. This modifies the internal quality letter.
+    constexpr cigar & assign_count(uint32_t const c)
+    {
+        get<1>(*this) = c;
+        return *this;
+    }
+    //!\}
+
+    /*!\name Read functions
+     * \{
+     */
+    //!\brief Return a character. This reads the internal sequence letter.
+    constexpr char_type to_char() const noexcept
+    {
+        return seqan3::to_char(get<0>(*this));
+    }
+
+    //!\brief Return the count value. This reads the internal quality letter.
+    constexpr uint32_t to_count() const noexcept
+    {
+        return get<1>(*this);
+    }
+
+    //!\brief Return the count value. This reads the internal quality letter.
+    string_type to_string() const noexcept
+    {
+        std::string tmp{std::to_string(get<1>(*this))};
+        tmp.push_back(seqan3::to_char(get<0>(*this)));
+        return tmp;
+    }
+    //!\}
+};
+
+//!\brief Type deduction guide enables usage of cigar without specifying template args.
+//!\relates cigar
+template <typename cigar_type, typename count_type>
+cigar(cigar_type &&, count_type &&)
+    -> cigar<std::decay_t<cigar_type>, std::decay_t<count_type>>;
+
+/*!\brief Specialized implementation of seqan3::alphabet_concept::operator<<()
+ *        for cigar delegates to `cig.to_string()`.
+ * \relates cigar
+ * \param os  The output stream you are printing to.
+ * \param cig The seqan3::cigar letter that you wish to print.
+ * \returns A reference to the output stream.
+ */
+std::ostream & operator<<(std::ostream & os, cigar<> const & cig)
+{
+    os << cig.to_string();
+    return os;
+}
+
+} // namespace seqan3
+
+// ------------------------------------------------------------------
+// containers
+// ------------------------------------------------------------------
+
+namespace seqan3
+{
+
+//!\brief Alias for an std::vector of seqan3::cigar.
```

Comment:
of `seqan3::cigar<cigar_op, uint32_t>`
Because you use always the default template args here.